backend/metamodel/syson-sysml-metamodel/src/main/java/org/eclipse/syson/sysml/impl/MultiplicityRangeImpl.java

```diff
@@ -85,13 +93,25 @@ public Expression getLowerBound() {
     /**
      * <!-- begin-user-doc -->
      * <!-- end-user-doc -->
-     * @generated
+     * @generated NOT
      */
     public Expression basicGetLowerBound() {
-        // TODO: implement this method to return the 'Lower Bound' reference
-        // -> do not perform proxy resolution
-        // Ensure that you remove @generated or mark it @generated NOT
-        return null;
+        List<Expression> expressions = getOwnedElement().stream()
+                .filter(Expression.class::isInstance)
+                .map(Expression.class::cast)
+                .collect(Collectors.toList());
+
+        if (expressions.isEmpty()) {
+            return null;
+        }
+
+        if (expressions.get(0) instanceof LiteralInfinity) {
```

Comment:
I think the condition here is not complete.  This case is only true if there is 2 expression right?

I would rewrite this methode link this


```java
 List<Expression> expressions = getOwnedElement().stream()
                .filter(Expression.class::isInstance)
                .map(Expression.class::cast)
                .toList();
        
        final Expression lowerBound;
        if (expressions.isEmpty()) {
            lowerBound = null;
        } else if (expressions.size() == 1  && expressions.get(0) instanceof LiteralInfinity) {
            // Only upper bound specified and is infinity
            LiteralInteger literal = SysmlFactoryImpl.eINSTANCE.createLiteralInteger();
            literal.setValue(0);
            lowerBound = literal;
        } else {
            lowerBound = expressions.get(0);
        }

        return lowerBound;
```

Reply:
thanks, done